### PR TITLE
Migrate to Swift 3.0

### PR DESCRIPTION
**Pull request by norio-nomura**

### CODE REVIEW

**Comment by norio-nomura** [comment]
Failing tests are affected by https://github.com/jpsim/SourceKitten/issues/269.

**Comment by norio-nomura** [comment]
Updated to 0.13.0

**Comment by jpsim** [comment]
@norio-nomura how can I help you with this?

**Comment by norio-nomura** [comment]
@jpsim Maybe we should file PRs against `swift3.0` branch based on each changes that have been introduced into master by recently merged PRs.
Could you please review them and merge to `swift3.0` branch?

By the way, I have lost my push permission to realm/SwiftLint three days ago. Should I push them to my forked repositories?

**Comment by jpsim** [comment]
> By the way, I have lost my push permission to realm/SwiftLint three days ago. Should I push them to my forked repository?

Oh, I'm sorry about that. I don't know why that happened. I just re-added you.

As for making PRs towards `swift3.0` for each change for the last two months... that seems like quite a bit of work, and it may be simpler to merge `master` into this and fix the merge conflicts?

I'm open to suggestions.

**Comment by norio-nomura** [comment]
> each change for the last two months

About four weeks. I have not reviewed changes since I cut `0.13.0`.

> it may be simpler to merge master into this and fix the merge conflicts?

Ok, I'll continue pushing commits to `swift3.0` branch that merging each PRs in last four weeks.

**Comment by norio-nomura** [comment]
Updated.

**Comment by jpsim** [comment]
Thanks for doing all this @norio-nomura. This is fully ready for review now?

**Comment by norio-nomura** [comment]
I removed [WIP], but tests related docs are still failing because of https://github.com/jpsim/SourceKitten/issues/269.

**Comment by norio-nomura** [comment]
`swift test` crashes on getting `CFBundleShortVersionString` from bundle.

**Comment by jpsim** [comment]
> tests related docs are still failing because of jpsim/SourceKitten#269.

That's true, but we have that limitation when using the current `master` version of SwiftLint with a Swift 3 codebase right now too, it's just that we don't test that configuration.

So I'd opt for disabling those tests.

---

Is there anything else that needs to go in this before we can merge? Every day we don't merge this just adds extra work to keep it updated with changes to `master`.

**Comment by norio-nomura** [comment]
> Is there anything else that needs to go in this before we can merge?

I'm trying to replace YamlSwift with Yams. But, that is not needed to be into this.
Let's merge. 👍

**Comment by jpsim** [comment]
Let's disable the failing docs tests before merging, you mentioned a failure with getting the version from the bundle in SPM too. Does that still happen? Hard to tell with Travis being so ridiculously backed up. It might as well be down

**Comment by norio-nomura** [comment]
Updated.

**Comment by jpsim** [comment]
Should we take care of Swift 3 API Guidelines in a later PR? I'm inclined to say yes.

**Comment by norio-nomura** [comment]
Yeah, as inspecting [search results](https://github.com/search?q=import+SwiftLintFramework&type=Code&utf8=✓) , those will affect to only us on github. 😉

**Comment by norio-nomura** [comment]
Maybe following would have higher priority than Swift 3 API Guidelines.

> Every day we don't merge this just adds extra work to keep it updated with changes to `master`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 ##### Breaking
 
-* None.
+* SourceKitten now requires Xcode 8.x and Swift 3.0 to build.
```
Comment (norio-nomura) [inline CHANGELOG.md]: ~SourceKitten~
SwiftLint

**Comment by jpsim** [comment]
Anything else @norio-nomura?

**Comment by norio-nomura** [comment]
@jpsim None!

**Comment by norio-nomura** [comment]
Oh, wait a minute.

**Comment by norio-nomura** [comment]
Done! 😃👍

**Comment by norio-nomura** [comment]
I have canceled some outdated jobs on travis.

**Comment by jpsim** [comment]
:shipit:

**Pull request merged**